### PR TITLE
chore(client): migrate TestSuite component to TypeScript

### PR DESCRIPTION
**Pull request by sitek94**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Is part of #43778

Migrated TestSuite component to TypeScript.

### CODE REVIEW

File: client/src/redux/prop-types.ts [hunk 1]
```diff
@@ -258,7 +258,7 @@ export type DimensionsType = {
 export type Test = {
   pass?: boolean;
   err?: string;
-} & (ChallengeTest | CertTest);
+} & ChallengeTest;
```
[the remark this answers is not in the document]
Comment (raisedadead) [reply client/src/redux/prop-types.ts]: @ShaunSHamilton is this resolved and ready?

**Comment by moT01** [comment]
✅ rebased
✅ conflicts fixed

File: client/src/templates/Challenges/components/tool-panel.tsx [hunk 2]
```diff
@@ -30,7 +30,7 @@ interface ToolPanelProps {
   openVideoModal: () => void;
   openResetModal: () => void;
   guideUrl: string;
-  videoUrl: string;
+  videoUrl?: string;
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/components/tool-panel.tsx:33]:
```suggestion
  videoUrl: string;
```

File: client/src/templates/Challenges/components/side-panel.tsx [hunk 3]
```diff
@@ -22,8 +23,8 @@ interface SidePanelProps {
   guideUrl: string;
   instructionsPanelRef: React.RefObject<HTMLDivElement>;
   showToolPanel: boolean;
-  tests?: Record<string, unknown>[];
-  videoUrl: string;
+  tests?: Test[];
+  videoUrl?: string;
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/components/side-panel.tsx:27]:
```suggestion
  videoUrl: string;
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
The `videoUrl`s can be required (removing the ?s doesn't flag up any errors, so I think we're good)

**Review by ShaunSHamilton — APPROVED** [review]
Thanks, for your patience, @sitek94 .

I have just pushed the changes Oliver suggested, with one of my own.

This LGTM 👍 

cc @ojeytonwilliams

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1: 

Thanks @sitek94!

**Pull request merged**